pybossa/view/projects.py

```diff
@@ -2015,24 +2149,31 @@ def task_n_answers(short_name):
                         pro_features=pro)
         return handle_content_type(response)
     elif request.method == 'POST':
-        if default_form.validate():
-            project.set_default_n_answers(default_form.default_n_answers.data)
-            auditlogger.log_event(project, current_user, 'update', 'project.default_n_answers',
-                      'N/A', default_form.default_n_answers.data)
-            msg = gettext('Redundancy updated!')
-            flash(msg, 'success')
-        elif form.validate():
-            tasks_not_updated = task_repo.update_tasks_redundancy(project, form.n_answers.data)
-            if tasks_not_updated:
-                notify_redundancy_updates(tasks_not_updated)
-                flash('Redundancy of some of the tasks could not be updated. An email has been sent with details')
-            else:
+        exist_error = False
+        if default_form.default_n_answers.data is not None:
+            if default_form.validate():
+                project.set_default_n_answers(default_form.default_n_answers.data)
+                auditlogger.log_event(project, current_user, 'update', 'project.default_n_answers',
+                        'N/A', default_form.default_n_answers.data)
                 msg = gettext('Redundancy updated!')
                 flash(msg, 'success')
-            # Log it
-            auditlogger.log_event(project, current_user, 'update', 'task.n_answers',
-                                  'N/A', form.n_answers.data)
-        if default_form.validate() or form.validate():
+            else:
+                exist_error = True
+        if form.n_answers.data is not None:
+            if form.validate():
+                tasks_not_updated = task_repo.update_tasks_redundancy(project, form.n_answers.data)
+                if tasks_not_updated:
+                    notify_redundancy_updates(tasks_not_updated)
+                    flash('Redundancy of some of the tasks could not be updated. An email has been sent with details')
+                else:
+                    msg = gettext('Redundancy updated!')
+                    flash(msg, 'success')
+                # Log it
+                auditlogger.log_event(project, current_user, 'update', 'task.n_answers',
+                                    'N/A', form.n_answers.data)
+            else:
+                exist_error = True
+        if not exist_error:
```

Comment:
as we return from here, else can be removed and following lines can be reindented